tdms/src/interpolate.cpp

```diff
@@ -204,76 +192,53 @@ void interpolateFieldCentralE_TM( double ***Ex_yee, double ***Ey_yee, double ***
- *
- *i_l[in]  - lowest i index into the FDTD grid to evaluate the field at. Should be >= 2.
- *i_u[out] - largest i index into the FDTD grid to evaluate the field at. Should be <= I-2.
- *j_l[in]  - lowest i index into the FDTD grid to evaluate the field at. Should be >= 2.
- *j_u[out] - largest i index into the FDTD grid to evaluate the field at. Should be <= J-2.
- *k_l[in]  - lowest i index into the FDTD grid to evaluate the field at. Should be >= 2.
- *k_u[out] - largest i index into the FDTD grid to evaluate the field at. Should be <= K-2.
- *
- *(I - number of elements in the i direction of the FDTD grid
- * J - number of elements in the j direction of the FDTD grid
- * K - number of elements in the k direction of the FDTD grid)
+ * @param[in] Ex_yee Input Yee cell Ex field matrix
+ * @param[in] Ey_yee Input Yee cell Ey field matrix
+ * @param[in] Ez_yee Input Yee cell Ez field matrix
+ * @param[out] Ex Output Yee cell Ex field matrix
+ * @param[out] Ey Output Yee cell Ey field matrix
+ * @param[out] Ez Output Yee cell Ez field matrix
+ * @param[in] i_l Least i index into the FDTD grid to evaluate the field at. Should be >= 2
+ * @param[in] i_u Greatest i index into the FDTD grid to evaluate the field at. Should be <= I-2
+ * @param[in] j_l Least j index into the FDTD grid to evaluate the field at. Should be >= 2
+ * @param[in] j_u Greatest i index into the FDTD grid to evaluate the field at. Should be <= J-2
+ * @param[in] k_l Least k index into the FDTD grid to evaluate the field at. Should be >= 2
+ * @param[in] k_u Greatest i index into the FDTD grid to evaluate the field at. Should be <= K-2
+ * 
+ * I, J, K are the number of elements in the i, j, k directions of the FDTD grid, respectively.
  */
 void mxInterpolateFieldCentralE( mxArray *Ex_yee , mxArray *Ey_yee , mxArray *Ez_yee,
 				mxArray **Ex    , mxArray **Ey    , mxArray **Ez    ,
 				int i_l, int i_u, int j_l, int j_u, int k_l, int k_u){
 
   double ***ExR, ***ExI, ***EyR, ***EyI, ***EzR, ***EzI,***Ex_yee_R, ***Ex_yee_I, ***Ey_yee_R, ***Ey_yee_I, ***Ez_yee_R, ***Ez_yee_I;
   const int *indims;
-  int outdims[3], ndims;
+  int outdims[3];
+
   //fprintf(stderr, "mxInterpolateFieldCentralE Pos 00\n");
   if( (int)mxGetNumberOfDimensions( (const mxArray *)Ex_yee) < 3){
-    throw runtime_error("Error in mxInterpolateFieldCentralE, Ex_yee does not have 3 dimensions\n");
-    
-    
+    throw runtime_error("Error in mxInterpolateFieldCentralE, Ex_yee does not have 3 dimensions\n");   
   }
   //fprintf(stderr, "mxInterpolateFieldCentralE Pos 01\n");
+
   indims = (int *)mxGetDimensions( (mxArray *)Ex_yee);
   //fprintf(stderr, "mxInterpolateFieldCentralE(indims): (%d,%d,%d)\n",indims[0],indims[1],indims[2]);
   //fprintf(stderr, "mxInterpolateFieldCentralE: j_u: %d, j_l: %d\n",j_u,j_l);
-  //assume that all matrices have the same dimensions
+  //assume that all matrices have the same dimensions - DANGEROUS, SHOULDN'T WE CHECK THIS??
```

Comment:
> DANGEROUS, SHOULDN'T WE CHECK THIS??

I guess it depends how performance critical this function is.. my guess it not very as it exists outside of the main loop(?) in which case lets definitely add a check!

Reply:
this constraint can be implicit when we pass an `ElectricField` instance